runners/flink/src/test/java/org/apache/beam/runners/flink/streaming/UnboundedSourceWrapperTest.java

```diff
@@ -134,11 +133,13 @@ public void testValueEmission() throws Exception {
               public void run() {
                 while (true) {
                   try {
-                    synchronized (testHarness.getCheckpointLock()) {
-                      testHarness.setProcessingTime(System.currentTimeMillis());
-                    }
+                    testHarness.setProcessingTime(System.currentTimeMillis());
                     Thread.sleep(1000);
```

Comment:
Did you find why this sleep is here?

Reply:
Whenever processing time is advanced, the watermark timers are triggered, so we need to keep it advancing. The sleep releases the checkpoint lock which would otherwise prevent the the source wrapper from making progress.